### ISSUE TITLE
Rule proposal: `no-magic-array-flat-depth`

### ISSUE THREAD

**Issue by fisker** [sindresorhus/eslint-plugin-unicorn#1920]
### Description

A special case of [`no-magic-numbers`](https://eslint.org/docs/latest/rules/no-magic-numbers) rule.

When calling `Array#flat(depth)` with depth. Normally the depth should be `1`/`Number.POSITIVE_INFINITY`. Otherwise it should be reported, but I don't want this rule to get annoying, so

- If it's not a number literal, we assume user know the max depth.
- If there is a comment before the number, we assume the number is explained. 

### Fail

```js
array.flat(2);
```

```js
array.flat(99);
```

### Pass

```js
array.flat();
```

```js
array.flat(Number.POSITIVE_INFINITY);
```

```js
array.flat(depth);
```

```js
array.flat(/* The depth is always 2 */ 2);
```

### Additional Info

This rule should only be useful for users don't want to enable [`no-magic-numbers`](https://eslint.org/docs/latest/rules/no-magic-numbers), otherwise the failed cases shold already covered.

**Comment by sindresorhus** [comment]
Are there other APIs with a depth parameter? Otherwise, I would name it more explicitly: `no-magic-array-flat-depth`.

**Comment by fisker** [comment]
Can't think another one.

**Comment by fisker** [comment]
Accepted.
